app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -1,11 +1,31 @@
 package com.example.drawingactivity
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+
+// What do I need to do?
+// create a Dialog on Click and BottomSlide
+// Dialog contains Gradient, Color Info Views, and Hue Selector
+// Be able to click on the views and they update
+// be able to close and reopen the view with it retaining last state
+// be able to change orientation without losing state
+
+// will grab values from my instantiated classes here to pass into the viewModel
 
 class MainActivity : AppCompatActivity() {
+
+    private val viewModel = CustomColorViewModel()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        val colorSelector = findViewById<PrimaryColorCircleView>(R.id.colorCircle)
+
+        colorSelector.onChangeColor = { viewModel.setPrimaryColor() }
```

Comment:
The color selector should pass an `it` here that the viewmodel consumes, unless you want to name the viewmodel function `cyclePrimaryColor`. As it is they look unconnected

Reply:
renamed function to reflect what it does